lib/keyrod/fedcloud_client.rb

```diff
@@ -0,0 +1,122 @@
+require 'faraday'
+require 'json'
+require 'net/http'
+
+module Keyrod
+  class FedcloudClient
+    attr_reader :site, :access_token, :auth_path, :ssl
+
+    PROJECTS_PATH = '/v3/auth/projects'.freeze
+    SCOPED_PATH = '/v3/auth/tokens'.freeze
+    REDIRECT_HEADER = 'WWW-Authenticate'.freeze
+    REDIRECT_REGEXP = /(?<=\')(.*?)(?=\')/
+
+    def initialize
+      @site = Keyrod::Settings[:site]
+      @access_token = Keyrod::Settings[:'access-token']
+      @auth_path = "/v3/OS-FEDERATION/identity_providers/#{Keyrod::Settings[:'identity-provider']}/protocols/oidc/auth"
+      @ssl = Keyrod::Settings[:ssl]
+    end
+
+    def unscoped_token
+      conn = connection(unscoped_token_params)
+
+      logger.debug "unscoped_token: Sending request with headers #{conn.headers}"
+
+      response = handle_response(conn)
+
+      raise Keyrod::Errors::ResponseError, "Response for getting unscoped token was #{response.status}" unless response.success?
+      response.headers[:'X-Subject-Token']
+    end
+
+    def projects(unscoped_token)
+      conn = connection(projects_params(unscoped_token))
+
+      logger.debug "projects: Sending request with headers #{conn.headers}"
+
+      response = handle_response(conn)
+
+      raise Keyrod::Errors::ResponseError, "Response for getting list of projects was #{response.status}" unless response.success?
+      parse_projects(response.body)
+    end
+
+    def scoped_token(unscoped_token, project)
+      conn = connection(scoped_token_params)
+
+      logger.debug "scoped_token: Sending request with headers #{conn.headers}"
+
+      response = handle_response(conn, scoped_token_body(unscoped_token, project))
+
+      raise Keyrod::Errors::ResponseError, "Response for getting scoped token was #{response.status}" unless response.success?
+      response.headers[:'X-Subject-Token']
+    end
+
+    private
+
+    def unscoped_token_params(fc_site = site)
```

Comment:
I still don't see any reason to use that `fc_site` parameter. Am I missing something?